nowcasting_dataset/data_sources/gsp/eso.py

```diff
@@ -0,0 +1,78 @@
+import json
+import urllib
+import logging
+from typing import List
+from urllib.request import urlopen
+
+import geopandas as gpd
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+WGS84_CRS = "EPSG:4326"
+
+
+def get_gsp_metadata_from_eso() -> pd.DataFrame:
+    """
+    Get the metadata for the gsp, from ESO.
+    @return:
+    """
+
+    # call ESO website
+    url = (
+        "https://data.nationalgrideso.com/api/3/action/datastore_search?"
+        "resource_id=bbe2cc72-a6c6-46e6-8f4e-48b879467368&limit=400"
+    )
+    fileobj = urllib.request.urlopen(url)
+    d = json.loads(fileobj.read())
+
+    # make dataframe
+    results = d["result"]["records"]
+    metadata = pd.DataFrame(results)
+
+    # drop duplicates
+    return metadata.drop_duplicates(subset=['gsp_id'])
+
+
+def get_gsp_shape_from_eso() -> gpd.GeoDataFrame:
+    """
+    Get the the gsp shape file
+    """
+
+    logger.debug('Loading GSP shape file')
+
+    url = (
+        "https://data.nationalgrideso.com/backend/dataset/2810092e-d4b2-472f-b955-d8bea01f9ec0/resource/"
+        "a3ed5711-407a-42a9-a63a-011615eea7e0/download/gsp_regions_20181031.geojson"
+    )
+
+    with urlopen(url) as response:
+        return gpd.read_file(response).to_crs(WGS84_CRS)
+
+
+def get_list_of_gsp_ids(maximum_number_of_gsp: int) -> List[int]:
+    """
+    Get list of gsp ids from ESO metadata
```

Comment:
Should we standardise on using Google's docstring style for our code (instead of Sphinx)?  Personally, I find Google's style easier to read in the code itself :)

Reply:
Yea, I've now changed my PyCharm to do Google docstrings, but this must have been before that. I'll try and re-do some obvious ones.